app/uk/gov/hmrc/cdsreimbursementclaimfrontend/controllers/claims/EnterAssociatedMrnController.scala

```diff
@@ -65,81 +85,139 @@ class EnterAssociatedMrnController @Inject() (
       )(identity)(Some(_))
     )
 
-  def enterMrn(index: AssociatedMrnIndex): Action[AnyContent] = authenticatedActionWithSessionData { implicit request =>
-    Ok(enterAssociatedMrnPage(index, mrnInputForm()))
-  }
+  def changeMrn(index: AssociatedMrnIndex): Action[AnyContent] =
+    enterMrn(index)
 
-  def submitEnteredMrn(index: AssociatedMrnIndex): Action[AnyContent] =
+  def enterMrn(index: AssociatedMrnIndex): Action[AnyContent] =
     authenticatedActionWithSessionData.async { implicit request =>
-      withAnswers[AssociatedMRNsAnswer] { (fillingOutClaim, associatedMRNs) =>
-        def showError: Form[AssociatedMrn] => Future[Result] =
-          formWithErrors => BadRequest(enterAssociatedMrnPage(index, formWithErrors))
-
-        def addMrn(): AssociatedMrn => EitherT[Future, Error, Unit] = mrn =>
-          EitherT {
-            val references = associatedMRNs.map(_ :+ mrn) orElse AssociatedMRNsAnswer(mrn).some
-            val claimCopy  = FillingOutClaim.of(fillingOutClaim)(_.copy(associatedMRNsAnswer = references))
-            updateSession(sessionStore, request)(_.copy(journeyStatus = claimCopy.some))
+      withAnswers[AssociatedMRNsAnswer] { (_, associatedMRNsAnswer) =>
+        associatedMRNsAnswer
+          .get(index)
+          .fold {
+            if (associatedMRNsAnswer.canAppendAt(index))
```

Comment:
Also another problem I had was that URL. Please can you do experiment - having `change-movement-reference-number/{n}` URL (when edit MRN) leave MRN field empty to get validation error. Would you see the same URL? If so then it is great, otherwise maybe you want to have a single `enter-movement-reference-number/{n}`?